patches/tModLoader/Terraria/ModLoader/ModExtraJumpLoader.cs

```diff
@@ -0,0 +1,220 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.Audio;
+using Terraria.ID;
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// This is where all ModExtraJump logic is handled and hooks called.
+	/// </summary>
+	public static class ModExtraJumpLoader
+	{
+		private static readonly IList<ModExtraJump> extraJumps = new List<ModExtraJump>();
+
+		internal static void SetNextJump(Player player, ref bool basilisk, ref bool goat, ref bool santank, ref bool unicorn, ref bool sandstorm, ref bool blizzard, ref bool fart, ref bool sail, ref bool cloud, ref bool anyModExtraJump) {
+			// This method is only called if atleast one modded jump exists
+			foreach (var extraJump in player.modExtraJumps) {
+				if (extraJump.index == VanillaExtraJump.Mounts.index) {
+					if (player.canJumpAgain_Basilisk) {
+						basilisk = true;
+						player.canJumpAgain_Basilisk = false;
+					}
+
+					if (player.canJumpAgain_WallOfFleshGoat) {
+						goat = true;
```

Comment:
This all should be made OOP in the actual non-lazy implementation. If we won't have hardcoded messes like this, then it'll be fine.

Reply:
how would you deal with the 27 fields related to jumps? Turn them into properties pointing at the associated `DoubleJump` and the field?